### ISSUE TITLE
unresolved symbols in compojure macros

### ISSUE THREAD

**Issue by slipset** [borkdude/clj-kondo#583]
When destructuring like this:
```Clojure
{{id :id} :params
  system :system :as req}
```
clj-kondo flags `id` and `system` as unresolved symbols. I would expect it to not.
![Screenshot 2019-11-05 at 13 19 58](https://user-images.githubusercontent.com/5894926/68207479-4e63fe80-ffcf-11e9-8817-c600e13ea2b7.png)

**Comment by borkdude** [comment]
@slipset Thanks! Could you provide a complete example, like:

foo.clj:
```
...
```
so I can do: 
```
clj-kondo --lint foo.clj
```
to reproduce it?

**Comment by slipset** [comment]
```clojure
(ns foo.core
  (:require [compojure.core :refer [ANY context routes]])
  (:gen-class))

(ANY "foo/:id" {{id :id} :params
                system :system :as req}
  id)
```

fails with 
```
14:16 $ clj-kondo --lint core.clj
core.clj:2:41: warning: #'compojure.core/context is referred but never used
core.clj:2:49: warning: #'compojure.core/routes is referred but never used
core.clj:5:18: error: unresolved symbol id
core.clj:6:17: error: unresolved symbol system
core.clj:6:36: error: unresolved symbol req
linting took 7ms, errors: 3, warnings: 2
✘-3 /tmp/foo/src/foo
```

**Comment by borkdude** [comment]
@slipset This is because `ANY` is a compojure macro. Clj-kondo doesn't understand custom macros, but you can teach/silence it using configuration.

I plan to include built-in compojure config (as part of #559), but for now you can use:

``` clojure
{:lint-as {compojure.core/defroutes clojure.core/def}
 :unresolved-symbol {:exclude [(compojure/ANY)]}}
```
in your `.clj-kondo/config.edn`.

**Comment by borkdude** [comment]
Or even better yet, clj-kondo add built-in support for the compojure macros, as a second pass on this.

**Comment by borkdude** [comment]
@slipset If I were to support the compojure macros out of the box by doing actual analysis, would I need to parse the routing string or are all relevant symbols introduced by the following destructuring form?

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/e5f8af54590b084cc119263698bdc786bfc6434f.
